src/input/Input.tsx

```diff
@@ -23,26 +23,27 @@ const renderText = (content, defaultProps, style) =>
     style: StyleSheet.flatten([style, defaultProps && defaultProps.style]),
   });
 
-export type InputProps = TextInputProps & {
-  containerStyle?: StyleProp<ViewStyle>;
-  disabled?: boolean;
-  disabledInputStyle?: StyleProp<TextStyle>;
-  inputContainerStyle?: StyleProp<ViewStyle>;
-  leftIcon?: any;
-  leftIconContainerStyle?: StyleProp<ViewStyle>;
-  rightIcon?: any;
-  rightIconContainerStyle?: StyleProp<ViewStyle>;
-  inputStyle?: object | any[];
-  InputComponent?: typeof React.Component;
-  errorProps?: object;
-  errorStyle?: object | any[];
-  errorMessage?: string;
-  label?: React.ReactNode;
-  labelStyle?: object | any[];
-  labelProps?: object;
-  theme?: Theme;
-  renderErrorMessage?: boolean;
-};
+export type InputProps = React.ComponentPropsWithRef<typeof TextInput> &
```

Comment:
This is closer to what I think is probably needed, but it still doesn't seem correct. It might be that there is some additional `forwardRef` that is needed to make this work, but I'm unsure about this. Did you get this idea from a particular article or website?

Reply:
Actually almost similar issue was there on react-native-paper. and the PR which solved this had done the exact same thing. I think we can go with this. I'll link the PR here

Reply:
At the very least `TextInputProps &` seems unnecessary with this addition.

Reply:
@flyingcircle we can replace TextInputProps with React.ComponentPropsWithRef<typeof TextInput> and it will work fine! Shall i make a new PR for this Input and Image!?